hamster_lib/storage.py

```diff
@@ -845,20 +845,49 @@ def _start_tmp_fact(self, fact):
             self.store.logger.debug(_("New temporary fact started."))
         return fact
 
-    def stop_tmp_fact(self):
+    def stop_tmp_fact(self, end_hint=None):
         """
         Stop current 'ongoing fact'.
 
+        Args:
+            end_hint (datetime.timedelta, or datetime.datetime): Offset to be
+                added to ``datetime.datetime.now()`` when computing for
+                ``self.end``. If you want the computed ``end`` to be *before*
+                ``now()`` you can pass negative ``timedelta`` values.
+
         Returns:
             hamster_lib.Fact: The stored fact.
 
         Raises:
+            TypeError: If ``end_hint`` is not a ``datetime.dateime`` or
+                ``datetime.timedelta`` instance.
             ValueError: If there is no currently 'ongoing fact' present.
+            ValueError: If the computed end (due to the hint) is before
+                the facts start value.
         """
         self.store.logger.debug(_("Stopping 'ongoing fact'."))
+
+        if not ((end_hint is None) or isinstance(end_hint, datetime.datetime) or (
```

Comment:
I do not follow. ``None`` is a valid parameter value, indicating that no hint logic is to be applied.
Did I miss something?